webpack/scenes/AlternateContentSources/Create/ACSCreateWizard.js

```diff
@@ -38,6 +38,8 @@ const ACSCreateWizard = ({ show, setIsOpen }) => {
   const [productIds, setProductIds] = useState([]);
   const [productNames, setProductNames] = useState([]);
   const [currentStep, setCurrentStep] = useState(1);
+  const [urlValidated, setUrlValidated] = useState('default');
+  const [subPathValidated, setSubPathValidated] = useState('default');
```

Comment:
I know this may not be strictly in scope for this PR, but I would _really_ love to see these states converted to values computed at render instead. It just makes the code so much simpler. Highly recommend reading this https://beta.reactjs.org/learn/you-might-not-need-an-effect#how-to-remove-unnecessary-effects-how-to-remove-unnecessary-effects

Reply:
I had these as variables inside the URL and subpath step in the 1st iteration on this. However, we need the urlValidated and subPathValidated values from the child step in the parent component, i.e the wizard to decide if next button should be enabled/disabled.

Reply:
If we move the validation functions to the parent component, we no longer need the states or the effect! Like this:

```diff
diff --git a/webpack/scenes/AlternateContentSources/Create/Steps/ACSReview.js b/webpack/scenes/AlternateContentSources/Create/Steps/ACSReview.js
index 90846aaad5..132fd577dc 100644
--- a/webpack/scenes/AlternateContentSources/Create/Steps/ACSReview.js
+++ b/webpack/scenes/AlternateContentSources/Create/Steps/ACSReview.js
@@ -20,7 +20,7 @@ const ACSReview = () => {
         title={__('Review details')}
         description={
           <>
-            {__('Review the information below and click ')}<b>{__('Add')}</b>{__(' to add your source. ' +
+            {__('Review the information below and click ')}<strong>{__('Add')}</strong>{__(' to add your source. ' +
                 'To edit details in previous steps, click ')}<b>{__('Back')}</b>{__(' or any step on the left.')
             }
           </>
diff --git a/webpack/scenes/AlternateContentSources/Create/Steps/AcsUrlPaths.js b/webpack/scenes/AlternateContentSources/Create/Steps/AcsUrlPaths.js
index 27fca58061..a6ca331d36 100644
--- a/webpack/scenes/AlternateContentSources/Create/Steps/AcsUrlPaths.js
+++ b/webpack/scenes/AlternateContentSources/Create/Steps/AcsUrlPaths.js
@@ -13,50 +13,9 @@ import WizardHeader from '../../../ContentViews/components/WizardHeader';
 const AcsUrlPaths = () => {
   const {
     url, setUrl, subpaths, setSubpaths, verifySSL, setVerifySSL,
-    urlValidated, setUrlValidated, subPathValidated, setSubPathValidated,
+    urlValidated, subPathValidated,
   } = useContext(ACSCreateContext);
 
-  const isValidUrl = (urlString) => {
-    try {
-      const urlFromString = new URL(urlString);
-      return (urlFromString.protocol === 'https:' || urlFromString.protocol === 'http:' || urlFromString.protocol === 'file:');
-    } catch (e) {
-      return false;
-    }
-  };
-
-  const areSubPathsValid = (subpathsString) => {
-    try {
-      if (subpathsString === '') {
-        return true;
-      }
-      return subpathsString.split(',').every((subpath) => {
-        const trimmedPath = subpath.trim();
-        return (!trimmedPath.startsWith('/') && trimmedPath.endsWith('/'));
-      });
-    } catch (e) {
-      return false;
-    }
-  };
-
-  const handleURLChange = (newUrl, _event) => {
-    setUrl(newUrl);
-    if (isValidUrl(newUrl)) {
-      setUrlValidated('default');
-    } else {
-      setUrlValidated('error');
-    }
-  };
-
-  const handleSubpathsChange = (newSubpaths, _event) => {
-    setSubpaths(newSubpaths);
-    if (areSubPathsValid(newSubpaths)) {
-      setSubPathValidated('default');
-    } else {
-      setSubPathValidated('error');
-    }
-  };
-
   return (
     <>
       <WizardHeader
@@ -81,7 +40,7 @@ const AcsUrlPaths = () => {
             placeholder="https:// or file://"
             value={url}
             validated={urlValidated}
-            onChange={value => handleURLChange(value)}
+            onChange={setUrl}
           />
         </FormGroup>
         <FormGroup
@@ -95,7 +54,7 @@ const AcsUrlPaths = () => {
             placeholder="test/repo1/, test/repo2/,"
             value={subpaths}
             validated={subPathValidated}
-            onChange={(value) => { handleSubpathsChange(value); }}
+            onChange={setSubpaths}
             name="acs_subpath_field"
             id="acs_subpath_field"
             aria-label="acs_subpath_field"
```

Reply:
I kind of prefer the validation logic for the step component to stay encapsulated in the component itself instead of the parent. Thoughts?

Reply:
Those validation logic functions are actually completely independent. So you could just put them outside the component altogether.

Reply:
That makes sense..Will move the 2 validators out and reuse them in the edit modal as well so we don't get another review from Maria.. :)